sql-plugin/src/main/330+/scala/com/nvidia/spark/rapids/shims/AnsiUtil.scala

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.shims
+
+import java.time.DateTimeException
+
+import ai.rapids.cudf.{ColumnVector, ColumnView, DType, Scalar}
+import com.nvidia.spark.rapids.{Arm, BoolUtils, FloatUtils, GpuColumnVector}
+
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.types.DataType
+
+object AnsiUtil extends Arm {
+
+  /**
+   * Spark 330+ supports Ansi cast from float/double to timestamp
+   * if have Nan or Infinity, throw an exception
+   * if (Math.floor(x) > Long.Max && Math.ceil(x) < Long.min), throw an exception
+   */
+  def supportsAnsiCastFloatToTimestamp(): Boolean = true
+
+  def castFloatToTimestampAnsi(floatInput: ColumnView, toType: DataType): ColumnVector = {
+    if(floatInput.getType.equals(DType.FLOAT32)) {
+      withResource(floatInput.castTo(DType.FLOAT64)) { d =>
+        castDoubleToTimestampAnsi(d, toType)
+      }
+    } else {
+      castDoubleToTimestampAnsi(floatInput, toType)
+    }
+  }
+
+  private def castDoubleToTimestampAnsi(doubleInput: ColumnView, toType: DataType): ColumnVector = {
```

Comment:
`hasSideEffects` for GpuCast needs to be updated for this because it throws an exception in ANSI mode.

Reply:
Done